core/shard/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+extern crate chain;
+extern crate parking_lot;
+extern crate primitives;
+extern crate rand;
+#[macro_use]
+extern crate serde_derive;
+extern crate storage;
+
+use chain::{Block, Header};
+use primitives::hash::{CryptoHash, hash_struct};
+use primitives::types::{AuthorityMask, MerkleHash, MultiSignature, PartialSignature, ReceiptTransaction, SignedTransaction};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ShardBlockHeaderBody {
```

Comment:
Why do we need both `ShardBlockHeaderBody` and `ShardBlockHeader`?

Reply:
Because Body is what is used in ShardBlock and ShardBlockHeader and what hash we are computing.
ShardBlockHeader is an individual object of the header - e.g. what can be sent around and stored, contains signatures and etc.

Reply:
I suggest we do the following renaming:
`ShardBlockHeaderBody -> ShardBlockHeader`
`ShardBlockHeader -> SignedShardBlockHeader`